### PR TITLE
fix(ui): keep user logged in after refresh

### PR DESCRIPTION
**Pull request by gordonpn**
Previously, due to the nature of most state management libraries, page refresh/reload would clear the stores. Now, using `/checkAuth` endpoint on page reloads the app would keep the user logged in if their jwt is valid.

Added react-router which allows for faster/cleaner route changes and created protected route component.

Added logout functionality with the `/logout` endpoint.

Please test with common usage scenarios.

**Update**: We should probably consider using Formik to handle more complex forms in the future.

### CODE REVIEW

File: client/src/components/LogOutButton.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        description: 'You have logged out successfully',
+        status: 'success',
+        duration: 2000,
+        isClosable: true,
+      });
+    } catch {
+      toast({
+        position: 'top',
+        title: 'An error occurred.',
+        description: 'Unable to log out',
```
Comment (Mackbellemore) [inline client/src/components/LogOutButton.jsx:32]: I feel like we should handle this better, If you clear your cookies and try to logout, we should still logout. I think we should check the status code on the error, if its a 403 then the user should still be logged out

Comment (Mackbellemore) [reply client/src/components/LogOutButton.jsx:32]: or we had the /user/logout route in the non authenticated paths on the server

Comment (gordonpn) [reply client/src/components/LogOutButton.jsx:41]: You're right, I just pushed a commit to check for 403 in the error. let me know what you think

**Review by Mackbellemore — APPROVED** [review]
:last_quarter_moon_with_face:

**Review by DavidLiang01 — APPROVED** [review]

**Pull request merged**